frontend/src/lib/components/ConfirmUpgradeModal/ConfirmUpgradeModal.tsx

```diff
@@ -0,0 +1,53 @@
+import { IconCheckCircle } from '@posthog/icons'
+import { LemonButton, LemonModal, Tooltip } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+
+import { confirmUpgradeModalLogic } from './confirmUpgradeModalLogic'
+
+export function ConfirmUpgradeModal(): JSX.Element {
+    const { upgradePlan } = useValues(confirmUpgradeModalLogic)
+    const { hideConfirmUpgradeModal, confirm, cancel } = useActions(confirmUpgradeModalLogic)
+
+    return (
+        <LemonModal
+            onClose={hideConfirmUpgradeModal}
+            isOpen={!!upgradePlan}
+            title={`Ready to subscribe to the ${upgradePlan?.name}?`}
+            footer={
+                <>
+                    <LemonButton type="secondary" onClick={() => cancel()}>
+                        Cancel
+                    </LemonButton>
+                    <LemonButton type="primary" onClick={() => confirm()}>
+                        Sign me up
+                    </LemonButton>
+                </>
+            }
+        >
+            <div className="max-w-120">
+                <p>
+                    Woo! You're gonna love the {upgradePlan?.name}. We're just confirming that this is a $
+                    {Number(upgradePlan?.unit_amount_usd)}/{upgradePlan?.unit} subscription and the first payment will
```

Comment:
The amount they are charged today is actually a prorated rate based on the remaining time in their billing period. So we need to calculate the time remaining in their billing period (if they have an active sub) and then go from there to get the approximate prorated amount.

Reply:
Do we have that calculation existing any where on the client already?

Reply:
I think it used to be in the pricing comparison cable but I took a quick peek there and didn't see it. But it's worth a deeper look.